server/api.go

```diff
@@ -260,6 +261,49 @@ func (p *Plugin) handleEndCall(w http.ResponseWriter, r *http.Request, channelID
 	res.Msg = "success"
 }
 
+func (p *Plugin) handleDismissNotification(w http.ResponseWriter, r *http.Request, channelID string) {
+	var res httpResponse
+	defer p.httpAudit("handleEndCall", &res, w, r)
+
+	userID := r.Header.Get("Mattermost-User-Id")
+
+	state, err := p.kvGetChannelState(channelID)
+	if err != nil {
+		res.Err = err.Error()
+		res.Code = http.StatusInternalServerError
+		return
+	}
+
+	if state == nil || state.Call == nil {
+		res.Err = "no call ongoing"
+		res.Code = http.StatusBadRequest
+		return
+	}
+
+	callID := state.Call.ID
+
+	if err := p.kvSetAtomicChannelState(channelID, func(state *channelState) (*channelState, error) {
+		if state == nil || state.Call == nil {
+			return nil, nil
+		}
+
+		if state.Call.ID != callID {
+			return nil, fmt.Errorf("previous call has ended and new one has started")
+		}
+
+		state.Call.DismissedNotification = append(state.Call.DismissedNotification, userID)
```

Comment:
I was referring to that line specifically, using an array we could be appending infinite number of duplicates if the endpoint gets called again without any deduplication check. Not a problem any longer since we are using a map now :+1: